### PR TITLE
feat(@vtmn/react): add alert component

### PR DESCRIPTION
**Pull request by AntoineDeloy**
## Pull Request checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch**. Don't request your main!
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.
- [x] Check PR name, it must follow the https://www.conventionalcommits.org pattern
- [x] If it's a new component in CSS, ask for design review

## Description

Pull Request to add the React alert component

## Other information

Tried to make the component a easy to use as possible, the intelligence of showing / hiding the alert needs to be in the application that uses this component and defined by the dev, the storybook shows the standard behaviour of this component

❤️ Thank you!

### CODE REVIEW

File: packages/sources/react/src/components/VtmnAlert/VtmnAlert.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+          `vtmn-alert_variant--${type}`,
+          { show: showAlert },
+          className,
+        )}
+      >
+        <article className="vtmn-alert_content">
+          <div className="vtmn-alert_content-title">
+            {title}
+            {onClose && (
+              <span className="vtmx-close-line" onClick={() => onClose} />
```
Comment (bastienterrier) [inline packages/sources/react/src/components/VtmnAlert/VtmnAlert.tsx:57]: Don't we want to call the `onClose` function?
I would have write `onClick={() => onClose}` or `onClick={onClose}` as we don't need `this` context

File: packages/sources/react/src/components/VtmnAlert/VtmnAlert.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import { VtmnAlertType } from './types';
+import clsx from 'clsx';
+
+export interface VtmnAlertProps
+  extends React.ComponentPropsWithoutRef<'dialog'> {
+  /**
+   * The alert's title
+   */
+  title: string;
```
Comment (bastienterrier) [inline packages/sources/react/src/components/VtmnAlert/VtmnAlert.tsx:10]: Is the `title` mandatory? I have a use-case when I don't want to show title, only some content. 
Is it necessary for accessibility purpose?

File: packages/sources/react/src/components/VtmnAlert/VtmnAlert.tsx (hunk 1 above)
Comment (AntoineDeloy) [reply packages/sources/react/src/components/VtmnAlert/VtmnAlert.tsx]: You're right !

**Comment by AntoineDeloy** [comment]
For me, the close icon should not be optional, and be always visible as the VtmnAlet component should always be closable.
Why ? 
An Alert should not be intrusive, the user should not be forced to wait the end of the alert in any way, and should be able to close it as soon as he clicks anywhere ON the component, the close icon would always be here just for the information of "okay, this component is closable if I want it to"

**Comment by lauthieb** [comment]
Thanks a lot @AntoineDeloy for this PR!
We will try to see why the `onClose` method is not called later on Storybook.
Have a nice evening.

**Comment by AntoineDeloy** [comment]
Still some PR to come don't worry !

**Pull request merged**